backends/backend_libkcapi.c

```diff
@@ -0,0 +1,2365 @@
+        ret = kcapi_pad_iv(handle, data->iv.buf, data->iv.len, &newiv, &newivlen);
+        if (ret)
+        {
+                logger(LOGGER_ERR, "gcm: iv padding failed\n");
+                goto out;
+        }
+
+        ret = -ENOMEM;
+        outbuflen = kcapi_aead_outbuflen_dec(handle, datalen, assoclen, taglen);
+        inbuflen = kcapi_aead_inbuflen_dec(handle, datalen, assoclen, taglen);
+
+        buf = calloc(1, inbuflen);
+        if (!buf)
+        {
+                logger(LOGGER_ERR, "gcm: allocation of buf failed\n");
+                goto out;
+        }
+
+        kcapi_aead_getdata_input(handle, buf, inbuflen, 0,
+                                        &assoc, &assoclen, &o_data, &datalen,
+                                        &tag, &taglen);
+
+        /* Set key */
+        if (!data->key.len ||
+                        !data->key.buf ||
+                        kcapi_aead_setkey(handle, data->key.buf, data->key.len))
+        {
+                logger(LOGGER_ERR, "gcm: symmetric cipher setkey failed\n");
+                goto out;
+        }
+
+        if(data->assoc.buf)
+                memcpy(assoc, data->assoc.buf, assoclen);
+        if(data->data.buf)
+                memcpy(o_data, data->data.buf, datalen);
+        if (data->tag.buf)
+                memcpy(tag, data->tag.buf, taglen);
+
+        /* Performing decryption */
+        ret = kcapi_aead_decrypt(handle, buf, inbuflen,
+                                                newiv,
+                                                buf, outbuflen,
+                                                KCAPI_ACCESS_SENDMSG);
+
+        if (ret < 0 && ret != -EBADMSG)
+        {
+                logger(LOGGER_ERR, "gcm: cipher operation of buffer failed: %d %d\n", errno, ret);
+                goto out;
+        }
+        else if (ret == -EBADMSG)
+        {
+                logger(LOGGER_DEBUG, "gcm: EBADMSG\n");
+                data->integrity_error = 1;
+                ret = 0;
+                goto out;
+        }
+        else if ((uint32_t)ret != outbuflen)
+        {
+                logger(LOGGER_ERR, "gcm: received data length %d does not match expected length %u\n", ret, outbuflen);
+                goto out;
+        }
+
+        if(data->iv.buf)
+                free_buf(&(data->iv));
+        if(data->data.buf)
+                free_buf(&(data->data));
+        if(data->tag.buf)
+                free_buf(&(data->tag));
+
+        CKINT_LOG(alloc_buf(datalen, &data->data), "gcm: pt buffer could not be allocated\n");
+        if(o_data)
+                memcpy(data->data.buf, o_data, datalen);
+
+out:
+        kcapi_aead_destroy(handle);
+        if (newiv)
+                free(newiv);
+        if (buf)
+                free(buf);
+        return ret;
+}
+
+static int ccm_encrypt(struct aead_data *data, flags_t parsed_flags)
+{
+        struct kcapi_handle *handle = NULL;
+        char cipher[] = "ccm(aes)";
```

Comment:
As discussed, please use existing function to return the name, instead of hardcoding.